src/Raven.Server/Documents/Handlers/AttachmentHandler.cs

```diff
@@ -0,0 +1,207 @@
+﻿// -----------------------------------------------------------------------
+//  <copyright file="DocumentHandler.cs" company="Hibernating Rhinos LTD">
+//      Copyright (c) Hibernating Rhinos LTD. All rights reserved.
+//  </copyright>
+// -----------------------------------------------------------------------
+
+using System.IO;
+using System.Net;
+using System.Runtime.ExceptionServices;
+using System.Threading.Tasks;
+using Raven.Client;
+using Raven.Client.Documents.Commands.Batches;
+using Raven.Server.Routing;
+using Raven.Server.ServerWide.Context;
+using Sparrow.Json;
+using Voron.Exceptions;
+
+namespace Raven.Server.Documents.Handlers
+{
+    public class AttachmentHandler : DatabaseRequestHandler
+    {
+        [RavenAction("/databases/*/attachments", "GET")]
+        public async Task Get()
+        {
+            var documentId = GetQueryStringValueAndAssertIfSingleAndNotEmpty("id");
+            var name = GetQueryStringValueAndAssertIfSingleAndNotEmpty("name");
+
+            DocumentsOperationContext context;
+            using (ContextPool.AllocateOperationContext(out context))
+            using (context.OpenReadTransaction())
+            {
+                var attachment = Database.DocumentsStorage.GetAttachment(context, documentId, name);
+                if (attachment == null)
+                {
+                    HttpContext.Response.StatusCode = (int)HttpStatusCode.NotFound;
+                    return;
+                }
+
+                var etag = GetLongFromHeaders("If-None-Match");
+                if (etag == attachment.Etag)
+                {
+                    HttpContext.Response.StatusCode = (int)HttpStatusCode.NotModified;
+                    return;
+                }
+
+                HttpContext.Response.Headers[Constants.Headers.Etag] = "\"" + attachment.Etag + "\"";
+
+                // Ensure that files are not cached at the browser side.
+                HttpContext.Response.Headers["Cache-Control"] = "no-cache, no-store, must-revalidate";
```

Comment:
Why is this required? The etag should handle that, no? How is this different from any other endpoint?